contrib/epee/include/net/levin_protocol_handler_async.h

```diff
@@ -495,7 +495,7 @@ class async_protocol_handler
           if(is_response)
           {//response to some invoke 
 
-            epee::critical_region_t<decltype(m_invoke_response_handlers_lock)> invoke_response_handlers_guard(m_invoke_response_handlers_lock);
+            boost::unique_lock invoke_response_handlers_guard(m_invoke_response_handlers_lock);
```

Comment:
what kind of c++17 voodoo is this! Who needs template parameters where we're going.